test/graphQl.js

```diff
@@ -0,0 +1,175 @@
+"use strict";
+var assert = require("assert");
+var jsonApiTestServer = require("../example/server");
+
+var Lokka = require('lokka').Lokka;
+var Transport = require('lokka-transport-http').Transport;
+var client = new Lokka({
+  transport: new Transport('http://localhost:16006/graphql')
+});
+
+
+describe("Testing jsonapi-server graphql", function() {
+
+  describe("read operations", function() {
+
+    it("filter with primary join and filter", function() {
+      return client.query(`
```

Comment:
The template literals and promises mean we no longer support node v0.10, so this work will have have to form a part of `v2.0.0`. I'll take the time to bump all our dependencies (notably Joi, which has a dependency on later versions of node) and sort out the linting at the same time.